services/galley/test/integration/API/MLS.hs

```diff
@@ -1630,3 +1632,81 @@ testBackendRemoveProposalLocalConvRemoteLeaver = do
         for_ bobClients $ \(_, ref) ->
           WS.assertMatch_ (5 # WS.Second) wsA $
             wsAssertBackendRemoveProposal bob qcnv ref
+
+-- for_ kprefs $ \kp ->
+--   WS.assertMatch_ (5 # WS.Second) wsA $ \notification ->
+--     void $
+--       wsAssertBackendRemoveProposal (pUserId bob) conversation kp notification
+
+testBackendRemoveProposalLocalConvLocalClient :: TestM ()
+testBackendRemoveProposalLocalConvLocalClient = do
+  [alice, bob] <- createAndConnectUsers [Nothing, Nothing]
+
+  runMLSTest $ do
+    [alice1, bob1, bob2] <- traverse createMLSClient [alice, bob, bob]
+    traverse_ uploadNewKeyPackage [bob1, bob2]
+    (_, qcnv) <- setupMLSGroup alice1
+    void $ createAddCommit alice1 [bob] >>= sendAndConsumeCommit
+    [(bobA, kpBobA), (bobB, _kpBobB)] <- getClientsFromGroupState alice1 bob
```

Comment:
Just for readability purposes, I propose to add assertions `bobA == bob1` and `bobB == bob2`.

Reply:
This assumption can't be made. I went with a solution involving `find`

Reply:
I am not sure I follow. I made this change locally on top of the PR:
```diff
diff --git a/services/galley/test/integration/API/MLS.hs b/services/galley/test/integration/API/MLS.hs
index 0927979dc..d562d1f2c 100644
--- a/services/galley/test/integration/API/MLS.hs
+++ b/services/galley/test/integration/API/MLS.hs
@@ -1613,6 +1613,8 @@ testBackendRemoveProposalLocalConvLocalClient = do
     (_, qcnv) <- setupMLSGroup alice1
     void $ createAddCommit alice1 [bob] >>= sendAndConsumeCommit
     [(bobA, kpBobA), (bobB, _kpBobB)] <- getClientsFromGroupState alice1 bob
+    liftIO $ assertEqual "bob1 /= bobA" bob1 bobA
+    liftIO $ assertEqual "bob2 /= bobB" bob2 bobB
```
and the test passes. These are assumptions I read the test with, and these are just codified versions of the assumptions.

Reply:
What I mean is that the order of `getClientsFromGroupState` is not guaranteed to be the order of generation. It might work, but I wouldn't trust it. For the test it's arbitraty whether `bob1` or `bob2` is deleted. But in any case with the latest changes I can pick `bob1` and have a less confusing test

Reply:
> What I mean is that the order of `getClientsFromGroupState` is not guaranteed to be the order of generation. It might work, but I wouldn't trust it. For the test it's arbitraty whether `bob1` or `bob2` is deleted. But in any case with the latest changes I can pick `bob1` and have a less confusing test

Oh, thanks for clarifying this! I wish we had `Map ClientIdentity KeyPackageRef` in the return type of `getClientsFromGroupState` instead of `[(ClientIdentity, KeyPackageRef)]`, which would clearly tell there's no order to expect among the pairs of associated values.